llvm-general/Setup.hs

```diff
@@ -106,10 +111,13 @@ main = do
 
     confHook = \(genericPackageDescription, hookedBuildInfo) configFlags -> do
       llvmConfig <- getLLVMConfig configFlags
-
-      llvmCxxFlags <- do
-        l <- llvmConfig "--cxxflags"
-        return $ (words l) \\ (map ("-D"++) uncheckedHsFFIDefines)
+      rawLlvmCxxFlags <- fmap words (llvmConfig "--cxxflags")
```

Comment:
The operational changes here are good, as is using the clearer name, but you're using unnecessarily broad scopes for some symbols, which makes the code harder to understand.

Please go back to keeping all the code for computing llvmCxxFlags in a sub-do-block, and please move the definition of stdlibPrefix down into the definition of stdLib.